server/api/include/irods/rsGeneralUpdate.hpp

```diff
@@ -1,10 +1,14 @@
-#ifndef RS_GENERAL_UPDATE_HPP
-#define RS_GENERAL_UPDATE_HPP
+#ifndef IRODS_RS_GENERAL_UPDATE_HPP
+#define IRODS_RS_GENERAL_UPDATE_HPP
 
 #include "irods/rcConnect.h"
 #include "irods/rodsGeneralUpdate.h"
 
+[[deprecated("rsGeneralUpdate is deprecated and will be removed in a later release. "
+             "If you are reaching for this, consider revising your approach.")]]
 int rsGeneralUpdate( rsComm_t *rsComm, generalUpdateInp_t *generalUpdateInp );
+
+// TODO Leading underscores are not allowed in the global namespace.
 int _rsGeneralUpdate( generalUpdateInp_t *generalUpdateInp );
```

Comment:
Sooo, C and C++ do not allow leading underscores in the global namespace and the last time I checked, iRODS uses leading underscores on functions everywhere.

Issue coming soon.

Reply:
See https://clang.llvm.org/extra/clang-tidy/checks/bugprone/reserved-identifier.html.

Reply:
Niiiiice

Reply:
Niiice

Reply:
Created #7579.

Removed TODO.